executor/read/step_executor.rs

```diff
@@ -180,10 +185,22 @@ pub(crate) fn create_executors_for_function(
     match &executable_function.returns {
         ExecutableReturn::Stream(positions) => {
             steps.push(StepExecutors::ReshapeForReturn(positions.clone()));
+            Ok(steps)
+        }
+        ExecutableReturn::Single(selector, positions) => {
+            steps.push(StepExecutors::ReshapeForReturn(positions.clone()));
+            let step = match selector {
+                SingleSelector::First => StreamModifierExecutor::new_first(PatternExecutor::new(steps)),
+                SingleSelector::Last => StreamModifierExecutor::new_last(PatternExecutor::new(steps)),
+            };
+            Ok(vec![step.into()])
+        }
+        ExecutableReturn::Check => todo!("ExecutableReturn::Check"),
```

Comment:
I could leave everything unimplemented, but I really wanted to test it with fetch. I'm not that interested in `Check`, so it's left with a todo.